### ISSUE TITLE
Enhanced localhost "homepage"

### ISSUE THREAD

**Issue by morvy** [leokhoa/laragon#22]
Hi Leo,

just wanted to share a little enhancement I'm using with Laragon as I have a lot of sites and I have bad memory, so this way it's easy to navigate to my sites.

Also I added mysql version check, but it needs credentials, so if you like it, maybe you'll find a way how to get db password automatically for this check..

https://gist.github.com/morvy/16a9bcba5543d44cc3453dceb1725494

**Comment by taskmann** [comment]
Hi Morvy,

I like this. However, although, laragon started MySQL, your version check give this line:

"MySQL not running!"

**Comment by morvy** [comment]
you need to have correct db credentials, if the connection cannot be made (server not running, server refused connection because of wrong credentials), it displays MySQL not running ..

**Comment by morvy** [comment]
@leokhoa I noticed that my script is not working correctly due to encoding of db password, could you point me to the right direction what you use for encoding the password saved in laragon.ini ? I see that 6A is "u" so it's not Hex to Ascii :D

**Comment by leokhoa** [comment]
@morvy : The password is encrypted to protect it from curious eyes (security best practice). In your case, you should consider removing MySQL password:
https://forum.laragon.org/topic/63/phpmyadmin-password-how-to-reset-mysql-root-s-password/3

**Comment by daugaard47** [comment]
@morvy  This is great. I'm seeing all my sites from my www folder, and I MySQL is running fine. The only main issue I'm having is in this line `<a href="'. $url .'://'. str_replace('{name}', $host, $laraconfig['HostnameFormat']). '">'.$favicon.' '.$host.'</a>` How would I open up my pretty URL's? 

Example: http://mysite.oo/ I'm currently getting taken to the following URL: about:blank#blocked

**Comment by daugaard47** [comment]
Never mind. I looked at the past commits. Got it working: `<a href="'. $url .'://'.$host.'.oo">'.$favicon.' '.$host.'</a>`
Thank you for this!

**Comment by morvy** [comment]
I forgot to update my gist, so you can copy new code if you want.
parse_ini_file had some issues on my PC too, so I replaced it with more "dumb" parsing.

Still, the part with parsing the password doesn't work, as Leo is encrypting it in laragon.ini and I don't know what should I do to test with encrypted password against DB ..

@leokhoa is it really needed to have that pass encrypted? because people have it in plaintext in all db configs in www folder + they probably know what they're setting as a password :)

**Comment by daugaard47** [comment]
@morvy I just forked it and edited it to my liking. Made darker theme buttons. https://gist.github.com/daugaard47/f6fb7a266541a81c308182f95ad5be7c
Is there a way to automatically get the Laragon version?
Wanted to do something like this: `Running on Laragon Full 4.0.11` But hard-coded this in.

**Comment by morvy** [comment]
I don't think that Laragon stores any information of current version somewhere. It's hardcoded in the exe file, even the exe displays wrong version in meta data, so no, I don't think it's possible

**Comment by alxndr-w** [comment]
If I didn't miss anything, this Issue was about enhancing the default webpage.

> as I have a lot of sites and I have bad memory

I don't know if you knew that this information is shown within Laragon UI:

![image](https://github.com/user-attachments/assets/f5cc5a63-40d8-4ae1-8621-ea5d3c741549)

Also, your website(s) will be shown here:

![image](https://github.com/user-attachments/assets/74a1c421-ea11-4e7c-888c-9a192616292a)

I know that, because my memory isn't always that good, too. ;)

That said, rethinking the main page could be done. Some developers prefer a dark theme, for example. 

But this should be discussed in a fresh issue: Which problem should be addressed? How can it be achieved? and then a Pull Request. 

Together with other discussions of enhancing the first page is the reason I will close this issue.
